### ISSUE TITLE
GetAlternateLookup<ReadOnlySpan<char>>() does not work on FrozenSet<string> in .NET 9 RC 1

### ISSUE THREAD

**Issue by mfeemster** [dotnet/runtime#107840]
### Description

I have installed .NET 9 RC 1 and have found that calling `GetAlternateLookup<ReadOnlySpan<char>>()` on `FrozenSet<string>` appears to not be supported. It gives an error of:

error CS9244: The type 'ReadOnlySpan<char>' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'TAlternate' in the generic type or method 'FrozenSet<string>.TryGetAlternateLookup<TAlternate>(out FrozenSet<string>.AlternateLookup<TAlternate>)'

### Reproduction Steps

```
var hs = new HashSet<string>()
{
	"str1",
	"str2",
	"str3",
}.ToFrozenSet(StringComparer.OrdinalIgnoreCase);
var alt = hs.GetAlternateLookup<ReadOnlySpan<char>>();//This line won't compile
```

### Expected behavior

This should compile because the documentation says it's been added to FrozenSet.

https://learn.microsoft.com/en-us/dotnet/api/system.collections.frozen.frozenset-1.getalternatelookup?view=net-9.0

### Actual behavior

The build fails with this error:

error CS9244: The type 'ReadOnlySpan<char>' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter 'TAlternate' in the generic type or method 'FrozenSet<string>.TryGetAlternateLookup<TAlternate>(out FrozenSet<string>.AlternateLookup<TAlternate>)'

### Regression?

No.

### Known Workarounds

None.

### Configuration

Visual Studio Community 2022  17.11.3
Windows 10
.NET 9 RC 1
`<LangVersion>preview</LangVersion>`

**Comment by svick** [comment]
I believe the problem here is that the reference assembly version of `HashSet<T>` is missing `where TAlternateKey : notnull, allows ref struct`.

Compare:

https://github.com/dotnet/runtime/blob/4c10effabb1ed66f98871ba7e89eaba7009507d1/src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenDictionary.AlternateLookup.cs#L24

https://github.com/dotnet/runtime/blob/4c10effabb1ed66f98871ba7e89eaba7009507d1/src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.net9.cs#L11

https://github.com/dotnet/runtime/blob/4c10effabb1ed66f98871ba7e89eaba7009507d1/src/libraries/System.Collections.Immutable/src/System/Collections/Frozen/FrozenSet.AlternateLookup.cs#L24

https://github.com/dotnet/runtime/blob/4c10effabb1ed66f98871ba7e89eaba7009507d1/src/libraries/System.Collections.Immutable/ref/System.Collections.Immutable.net9.cs#L25

**Comment by mfeemster** [comment]
This sounds correct. Do you know of a workaround? Or can it only be fixed when the next RC is released?

**Comment by stephentoub** [comment]
Thank you for reporting this.

> Do you know of a workaround? Or can it only be fixed when the next RC is released?

A hack would be to reference the implementation assembly rather than the ref assembly.

The only real fix is to fix the ref assembly, though.

https://github.com/dotnet/runtime/pull/107846
https://github.com/dotnet/runtime/pull/107847

**Comment by jeffhandley** [comment]
Fixed by #107846
